From 550196c9db2b76d9e23bc394f4dd02df78159a6a Mon Sep 17 00:00:00 2001
From: Peter Hawkins <phawkins@google.com>
Date: Wed, 3 Aug 2022 10:52:02 -0700
Subject: [PATCH 07/30] [NumPy] Fix NumPy 1.23 compatibility problems in
 tf_slim.

Multidimensional indexing by non-tuple values no longer works in NumPy 1.23 (https://numpy.org/devdocs/release/1.23.0-notes.html#expired-deprecations).

PiperOrigin-RevId: 465094836
Change-Id: I12f60917594827a2963360977fb849cfbbd5f749
---
 tf_slim/ops/variables.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tf_slim/ops/variables.py b/tf_slim/ops/variables.py
index d797355..0f988a9 100644
--- a/tf_slim/ops/variables.py
+++ b/tf_slim/ops/variables.py
@@ -665,7 +665,7 @@ def assign_from_checkpoint(model_path, var_list, ignore_missing_vars=False):
         slice_dims = zip(var._save_slice_info.var_offset,
                          var._save_slice_info.var_shape)
         slice_dims = [(start, start + size) for (start, size) in slice_dims]
-        slice_dims = [slice(*x) for x in slice_dims]
+        slice_dims = tuple(slice(*x) for x in slice_dims)
         slice_value = ckpt_value[slice_dims]
         slice_value = slice_value.reshape(var._save_slice_info.var_shape)
         feed_dict[placeholder_tensor] = slice_value
-- 
2.40.1

